lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -732,66 +733,157 @@ class MetricsEndpointProvider(Object):
     def __init__(
         self,
         charm,
-        name,
-        service_event,
+        relation_name=None,
         jobs=[],
-        alert_rules_path="src/prometheus_alert_rules",
+        alert_rules_path=None,
```

Comment:
What is wrong with a default path ? This was your own suggestion. What prompted the change here ?

Reply:
Because I realized that that default does not work in repos with an unconventional structure. It's documented in the PR description and commit message.